src/security/cert_generators/file/security_file_certgen.cc

```diff
@@ -184,7 +184,10 @@ static bool processNewRequest(Ssl::CrtdMessage & request_message, std::string co
     if (!request_message.parseRequest(certProperties, error))
         throw std::runtime_error("Error while parsing the crtd request: " + error);
 
-    Ssl::CertificateDb db(db_path, max_db_size, fs_block_size);
+    // TODO: create a DB object only once, instead re-allocating here on every call.
+    std::unique_ptr<Ssl::CertificateDb> db;
+    if (!db_path.empty() && max_db_size > 0)
```

Comment:
This is not a consistency check - it is the actual disk vs memory-only decision taking place. Since you required that the db become a pointer to flag later code operations we have to use the config settings to decide how to set the pointer up. It just looks weird and out of place because the CertificateDb object lifecycle is so screwed up.

Reply:
Using `-s` alone to decide whether to enable/use the database is probably the right approach.

Please make sure a lonely (i.e., without `-s`) `-M` is rejected.

I probably would not bother checking that `-M` is positive (when given together with `-s`). Printing a warning may be useful, but it can be considered out of this PR scope.

Reply:
done. This change also fixes the other issues around db_path and max_db_size conditions as side effects of the simplification.